src/main/java/seedu/duke/storage/Decoder.java

```diff
@@ -0,0 +1,96 @@
+package seedu.duke.storage;
+
+import seedu.duke.exercise.Exercise;
+import seedu.duke.exercise.ExerciseList;
+import seedu.duke.food.Food;
+import seedu.duke.food.FoodList;
+import seedu.duke.profile.Profile;
+import seedu.duke.profile.exceptions.InvalidCharacteristicException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+/**
+ * Converts the profile, exercise list and food list from storage and inputs into the bot.
+ */
+public class Decoder {
+    public static final String FILE_TEXT_DELIMITER = "\\|";
```

Comment:
Can consider adding message/exception to tell the user not use this delimiter when inserting any data :)

Reply:
Okie sure, will update the user guide respectively. Thanks rui yang 👍